### PR TITLE
Fixed a bug in XmlUTF8NodeWriter which caused EncoderFallbackException

### LINKED ISSUE #6076: [Port] System.Text.EncoderFallbackException: Unable to translate Unicode character
We fixed the issue on desktop. We need to port the fix to .Net Core.
```
  Discovering: System.Runtime.Serialization.Xml.Tests
  Discovered:  System.Runtime.Serialization.Xml.Tests
  Starting:    System.Runtime.Serialization.Xml.Tests
     XmlDictionaryWriterTest.XmlDictionaryWriter_InvalidUnicodeChar [FAIL]
        System.Text.EncoderFallbackException : Unable to translate Unicode character \uDB1B at index 0 to specified code page.
        Stack Trace:
              at System.Text.EncoderExceptionFallbackBuffer.Fallback(Char charUnknown, Int32 index)
              at System.Text.EncoderFallbackBuffer.InternalFallback(Char ch, Char*& chars)
              at System.Text.UTF8Encoding.GetByteCount(Char* chars, Int32 count, EncoderNLS baseEncoder)
              at System.Text.UTF8Encoding.GetByteCount(String chars)
              at System.Text.Encoding.GetBytes(String s)
           D:\OSS\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlStreamNodeWriter.cs(468,0): atSystem.Xml.XmlStreamNodeWriter.UnsafeGetUTF8Chars(Char* chars, Int32 charCount, Byte[] buffer, Int32 offset)
           D:\OSS\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlStreamNodeWriter.cs(354,0): atSystem.Xml.XmlStreamNodeWriter.UnsafeWriteUTF8Chars(Char* chars, Int32 charCount)
           D:\OSS\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextWriter.cs(480,0): at System.Xml.XmlUTF8NodeWriter.UnsafeWriteEscapedText(Char* chars, Int32 count)
           D:\OSS\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextWriter.cs(447,0): at System.Xml.XmlUTF8NodeWriter.WriteEscapedText(String s)
           D:\OSS\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBaseWriter.cs(1019,0): at System.Xml.XmlBaseWriter.WriteString(String value)
           D:\OSS\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlDictionaryAsyncCheckWriter.cs(381,0): at System.Xml.XmlDictionaryAsyncCheckWriter.WriteString(String text)
           D:\OSS\corefx\src\System.Runtime.Serialization.Xml\tests\XmlDictionaryWriterTest.cs(127,0): at XmlDictionaryWriterTest.XmlDictionaryWriter_InvalidUnicodeChar()
  Finished:    System.Runtime.Serialization.Xml.Tests

### PR DESCRIPTION
**Pull request by shmao**
The fix is ported form Desktop. Most of the fix had already been ported, but there still is a bug in the current code.

Fixed #6076

### CODE REVIEW

**Comment by shmao** [comment]
cc: @SGuyGe @khdang

**Comment by SGuyGe** [comment]
LGTM.

File: src/System.Private.DataContractSerialization/src/System/Xml/XmlUTF8TextWriter.cs [hunk 1]
```diff
@@ -101,7 +101,7 @@ public XmlUTF8NodeWriter(bool[] isEscapedAttributeChar, bool[] isEscapedElementC
         new public void SetOutput(Stream stream, bool ownsStream, Encoding encoding)
         {
             Encoding utf8Encoding = null;
-            if (encoding != null && encoding == Encoding.UTF8)
+            if (encoding != null && encoding.CodePage == Encoding.UTF8.CodePage)
```
Comment (khdang) [inline src/System.Private.DataContractSerialization/src/System/Xml/XmlUTF8TextWriter.cs]: Can we use encoding.Equals(Encoding.UTF8) for equality checking here? I think that includes the .CodePage equality check as well.

**Comment by khdang** [comment]
I have a comment regarding using encoding.Equals. LGTM.

**Comment by shmao** [comment]
Thanks for reviewing the PR. 
I still need to verify something with this change. I will merge this PR later.

File: src/System.Private.DataContractSerialization/src/System/Xml/XmlUTF8TextWriter.cs (hunk 1 above)
Comment (shmao) [inline src/System.Private.DataContractSerialization/src/System/Xml/XmlUTF8TextWriter.cs]: The Equals method also checks encoder/decoder fallback of the encoding instance besides checking the CodePage, which is different from this check. This is the behavior we have on Desktop. I'd prefer to keep the change as is for consistency.

**Pull request merged**